p2p/types/addr_book.go

```diff
@@ -0,0 +1,4 @@
+package types
+
+// AddrBook is a way of representing NetworkPeer sets
+type AddrBook []*NetworkPeer
```

Comment:
It's a light file, I don't mind lifting it :)
Honestly, I think it's gonna be worth splitting it when we start having more methods on these structs.

These two can definitely coexist, otherwise there are 1000 files were we have more than one struct,